apps/portal/pages/events/[slug]/qrcode.tsx

```diff
@@ -0,0 +1,22 @@
+import { useEffect, useRef } from 'react';
+import QRCode from 'qrcode';
+import { useRouter } from 'next/router';
+
+export default function QRCodePage() {
+  const { query } = useRouter();
+  const { slug } = query;
+
+  const canvasRef = useRef();
+
+  useEffect(() => {
+    if (canvasRef.current) {
+      QRCode.toCanvas(canvasRef.current, `/checkin/${slug}`, { margin: 3.5, width: 300 });
```

Comment:
Can you also append the base URL in addition to just `/checkin/${slug}` as well? For instance, it would be `https://next.portal.acmutd.co/checkin/${slug}` if in production and `http://localhost:4200/checkin/${slug}` if in development? Here's one way to extract the base URL from `getServerSideProps`: https://github.com/acmutd/hackportal/blob/develop/pages/index.tsx#L464-L469